ibm/service/scc/data_source_ibm_scc_posture_collectors.go

```diff
@@ -0,0 +1,526 @@
+						},
+						"use_private_endpoint": &schema.Schema{
+							Type:        schema.TypeBool,
+							Computed:    true,
+							Description: "Whether the collector should use a public or private endpoint. This value is generated based on is_public field value during collector creation. If is_public is set to true, this value will be false.",
+						},
+						"managed_by": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The entity that manages the collector.",
+						},
+						"trial_expiry": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The trial expiry. This holds the expiry date of registration_code. This field will be populated when collector is installed.",
+						},
+						"last_failed_internet_gateway_ip": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The failed internet gateway ip of the collector.",
+						},
+						"status_description": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The collector status.",
+						},
+						"reset_time": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The ISO Date/Time of the collector reset. This value will be populated when a collector is reset. The data-time when the reset event is occured is captured in this field.",
+						},
+						"is_public": &schema.Schema{
+							Type:        schema.TypeBool,
+							Computed:    true,
+							Description: "Determines whether the collector endpoint is accessible on a public network.If set to `true`, the collector connects to resources in your account over a public network. If set to `false`, the collector connects to resources by using a private IP that is accessible only through the IBM Cloud private network.",
+						},
+						"is_ubi_image": &schema.Schema{
+							Type:        schema.TypeBool,
+							Computed:    true,
+							Description: "Determines whether the collector has a Ubi image.",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceIBMSccPostureListCollectorsRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	postureManagementClient, err := meta.(conns.ClientSession).PostureManagementV2()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	listCollectorsOptions := &posturemanagementv2.ListCollectorsOptions{}
+
+	userDetails, err := meta.(conns.ClientSession).BluemixUserDetails()
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("[ERROR] Error getting userDetails %s", err))
+	}
+
+	accountID := userDetails.UserAccount
+	listCollectorsOptions.SetAccountID(accountID)
+
+	collectorList, response, err := postureManagementClient.ListCollectorsWithContext(context, listCollectorsOptions)
```

Comment:
This ListCollectorsWithContext handles pagination? If we have more than 50 or 100 will List give all or only first page?
If not so you need to add pagination  support

Reply:
We have pagination support. If no limit is given, it pulls only first page